### ISSUE TITLE
Canvas::into_inner(self)->Image returns a flipped image

### ISSUE THREAD

**Issue by Athorus** [ggez/ggez#693]
**Describe the bug**
When Canvas::into_inner(self)->Image is called, a flipped image is returned.

**To Reproduce**
See code below. 
[resources.zip](https://github.com/ggez/ggez/files/3747672/resources.zip) files is provided too (contains 2 png).

**Expected behavior**
The same image as the one displayed should be returned.

**Screenshots or pasted code**
The text at the bottom of the window is flipped. It should be the same as the text at the beginning of the window.
![canvas](https://user-images.githubusercontent.com/44225646/67157401-b0fc9f80-f32b-11e9-99b7-3655bea37b78.png)

```rust
use ggez::ContextBuilder;
use ggez::conf::Conf;
use ggez::Context;
use ggez::event::{self, EventHandler};
use ggez::GameResult;
use ggez::graphics::{self, Canvas, Color, DrawParam, Image};

use mint::Point2;

struct Application {
    dark_image: Image,
    dark_canvas: Canvas,
    light_image: Image,
    light_canvas: Canvas,
    canvas_image: Image
}

impl Application {
    fn new(ctx: &mut Context) -> Self {
        let (dark_image, dark_canvas) = Self::image_canvas(ctx, "/dark-image.png");
        let (light_image, light_canvas) = Self::image_canvas(ctx, "/light-image.png");

        let canvas_image = Self::canvas(ctx, &dark_image).into_inner();

        Self {dark_image, dark_canvas, light_image, light_canvas, canvas_image}
    }

    fn image_canvas(ctx: &mut Context, file: &str) -> (Image, Canvas) {
        let image = Image::new(ctx, file).unwrap();
        let canvas = Self::canvas(ctx, &image);
        (image, canvas)
    }

    fn canvas(ctx: &mut Context, image: &Image) -> Canvas {
        let canvas = Canvas::with_window_size(ctx).unwrap();
        graphics::set_canvas(ctx, Some(&canvas));
        graphics::draw(ctx, image, DrawParam::new()).unwrap();
        graphics::set_canvas(ctx, None);

        canvas
    }
}

impl EventHandler for Application {
    fn update(&mut self, _: &mut Context) -> GameResult<()> {
        Ok(())
    }

    fn draw(&mut self, ctx: &mut Context) -> GameResult<()> {
        graphics::clear(ctx, Color{r: 0., g: 0.3, b: 0., a: 1.});

        graphics::draw(ctx, &self.dark_image, DrawParam::new()).unwrap();
        let point2 = Point2 {x: 0., y: 60.};
        graphics::draw(ctx, &self.dark_canvas, DrawParam::new().dest(point2)).unwrap();

        let point2 = Point2 {x: 0., y: 200.};
        graphics::draw(ctx, &self.light_image, DrawParam::new().dest(point2)).unwrap();
        let point2 = Point2 {x: 0., y: 260.};
        graphics::draw(ctx, &self.light_canvas, DrawParam::new().dest(point2)).unwrap();

        graphics::draw(ctx, &self.canvas_image, DrawParam::new()).unwrap();

        graphics::present(ctx).unwrap();
        Ok(())
    }
}

pub fn main() {
    let c = Conf::new();
    let (ctx, event_loop) = &mut ContextBuilder::new("canvas", "athorus").conf(c).build().unwrap();
    let app = &mut Application::new(ctx);
    event::run(ctx, event_loop, app).unwrap();
}
```

**Hardware and Software:**
 - ggez version: 0.5.1
 - OS: Windows 10
 - Graphics card: AMD Radeon(TM) R9 290X
 - Graphics card drivers: Radeon 19.9.2

**Comment by icefoxen** [comment]
This is a known-ish issue, unfortunately... The ordering of lines and columns in GPU memory vs CPU memory is a little squirrelly, and ggez putting the origin at the top-left only makes it harder to understand.  Defining all the places where the order of lines needs to be flipped is tricky.

**Comment by nobbele** [comment]
Fixed by #936
